test/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubset.test.cpp

```diff
@@ -109,3 +80,80 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_State_CoordinatesSubset, Getter
         EXPECT_EQ(defaultSize_, defaultCoordinateSubset_.getSize());
     }
 }
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_State_CoordinatesSubset, Add)
+{
+    VectorXd input1(4);
+    VectorXd input2(4);
+
+    input1 << 1.0, 1.0, 2.0, 3.0;
+    input2 << 2.0, 7.0, 3.0, 4.0;
+
+    {
+        VectorXd expected(1);
+        VectorXd actual(1);
+
+        expected << 3.0;
+
+        actual = defaultCoordinateSubset_.add(
+            Instant::Undefined(), input1, input2, Frame::Undefined(), defaultCoordinatesBroker_
+        );
+
+        EXPECT_EQ(expected, actual);
+    }
+
+    {
+        VectorXd expected(3);
+        VectorXd actual(3);
+
+        expected << 8.0, 5.0, 7.0;
+
+        actual = defaultCartesianPosition_.add(
+            Instant::Undefined(), input1, input2, Frame::Undefined(), defaultCoordinatesBroker_
+        );
+
+        EXPECT_EQ(expected, actual);
+    }
```

Comment:
IMO this test should be in CartesianPosition.test.cpp
When possible we don't want to have children objects required for instantiation in the parent file.